lib/protocol/parser.js

```diff
@@ -0,0 +1,69 @@
+var msgpack = require('msgpack2');
+
+var constants = require('./constants');
+
+// Streaming data parser
+function Parser(callback, errback) {
```

Comment:
Callback and errback are usually used for promises where you can only have a single callback XOR errback associated as a return value. The parser gets some data and emits either messages or errors so an EventEmitter might make more sense.